pkg/controller/replication_controller.go

```diff
@@ -178,11 +205,13 @@ func (rm *ReplicationManager) watchControllers(resourceVersion *string) {
 	}
 }
 
-// Helper function. Also used in pkg/registry/controller, for now.
-func FilterActivePods(pods []api.Pod) []api.Pod {
+// filterActivePodsByNamespace returns all 'Active' pods in the given namespace.
+func filterActivePodsByNamespace(namespace string, pods []api.Pod) []api.Pod {
```

Comment:
I think that it's enough to return list of names. Copying whole objects is not necessary because we only use len() to add new pods and names to delete superfluous.